proxy/logging/MarshalIntegral.h

```diff
@@ -0,0 +1,247 @@
+/** @file
```

Comment:
I think that being able to unmarshal 64 bit values is probably sufficient for headers ATS can process and arbitrary sized values are a bit over kill.

Reply:
In spite of potential small performance hit particularly on non-64-bit platforms?

Reply:
ATS supports only 64 bit platforms. Even then, I'm not sure what performance hit you mean, other than marshaling 8 bytes instead of 4. If that's an issue, using a raw 32 bit value would be sufficient. Rather than having a class, just `memcpy` the four bytes and use `htonl` and `ntohl` if byte ordering is a concern.

Reply:
That could potentially imbed non-terminal nuls in the "string" value.  If embedded nuls are not a concern, should I drop the idea of storing the length of each tag and value?  I could just terminate them with a nul, and then have a double nul at the end of the stored header to indicate the end of the last tag/value pair in the header.